graphkit/network.py

```diff
@@ -214,9 +215,86 @@ def compute(self, outputs, named_inputs):
 
         # assert that network has been compiled
         assert self.steps, "network must be compiled before calling compute."
-        assert isinstance(outputs, (list, tuple)) or outputs == None,\
+        assert isinstance(outputs, (list, tuple)) or outputs is None,\
             "The outputs argument must be a list"
 
+
+        # choose a method of execution
+        if method == "parallel":
+            return self._compute_thread_pool_barrier_method(named_inputs,
+                                                            outputs)
+        else:
+            return self._compute_sequential_method(named_inputs,
+                                                   outputs)
+
+
+    def _compute_thread_pool_barrier_method(self, named_inputs, outputs,
+                                            thread_pool_size=10):
+        """
+        This method runs the graph using a parallel pool of thread executors.
+        You may achieve lower total latency if your graph is sufficiently
+        sub divided into operations using this method.
+        """
+        from multiprocessing.dummy import Pool
+
+        # if we have not already created a thread_pool, create one
+        if not hasattr(self, "_pool"):
```

Comment:
where could the `_pool` attribute have been initialized? will this always be initialized on the first run?

Reply:
Yea I was thinking of mainly doing this on first run because I wasn't able to think of a common use case where we'd want to have a different behavior.